photon-core/src/main/java/org/photonvision/common/dataflow/networktables/TimeSyncManager.java

```diff
@@ -128,9 +128,10 @@ synchronized void tick() {
         if (m_client != null) {
             var conns = ntInstance.getConnections();
 
-            if (conns.length > 0) {
-                logger.debug("Changing TimeSyncClient server to " + conns[0].remote_ip);
-                m_client.setServer(conns[0].remote_ip);
+            var newServer = conns[0].remote_ip;
+            if (conns.length > 0 && !m_client.getServer().equals(newServer)) {
```

Comment:
slightly confused / im bad at programming, why do we get the remote ip from the first element than check if there are things in the list? is this just guarenteed to have one element?

Reply:
totally bjork. a bug.